File: src/main/java/com/wafflestudio/draft/repository/RegionRepository.java
package com.wafflestudio.draft.repository;

import com.wafflestudio.draft.model.Region;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface RegionRepository extends JpaRepository<Region, Long> {

    Optional<Region> findById(Long id);

    List<Region> findByNameContaining(String name);
}

File: src/main/java/com/wafflestudio/draft/repository/RegionRepository.kt
package com.wafflestudio.draft.repository

import com.wafflestudio.draft.model.Region
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface RegionRepository : JpaRepository<Region?, Long?> {
    override fun findById(id: Long): Optional<Region?>
    fun findByNameContaining(name: String?): List<Region?>?
}